src/SignIn.js

```diff
@@ -14,17 +15,33 @@ class SignIn extends Component {
   state = {
     loggedIn: auth.loggedIn(),
     providers: [],
-    error: ''
+    error: '',
+    redirect: ''
+  };
+
+  endPoint() {
+    const baseUrl = "/sign-in";
+    return this.state.redirect ? `${baseUrl}?redirect=${this.state.redirect}` : baseUrl;
+  };
+
+  postAuthRedirect() {
+    return this.state.redirect ? `&postAuthRedirect=${this.state.redirect}` : "";
   };
 
   componentDidMount() {
+    const { redirect } = parse(window.location.search, { ignoreQueryPrefix: true });
+    if (redirect) this.setState({ redirect: redirect });
     fetch('/sign-in')
       .then(res => res.json())
       .then(data => this.setState(data));
   }
 
-  handleSuccess = (_, res) => {
+  handleSuccess = (data, res) => {
     auth.saveFromHeaders(res.headers);
+
+    if (data.redirect) {
+      window.location = data.redirect;
```

Comment:
In general, probably want to avoid setting `window.location` directly and use react-router. I think we can do something like this instead:

```js
this.props.history.push(data.redirect);
```